Modern/ops/src/main/java/org/bonej/ops/BoxIntersect.java

```diff
@@ -0,0 +1,174 @@
+
+package org.bonej.ops;
+
+import java.util.Optional;
+import java.util.stream.DoubleStream;
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.Op;
+import net.imagej.ops.special.function.AbstractBinaryFunctionOp;
+import net.imglib2.Interval;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.util.ValuePair;
+
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.vecmath.Tuple3d;
+import org.scijava.vecmath.Vector3d;
+
+/**
+ * Solves scalar values (t<sub>1</sub>, t<sub>2</sub>) for the parametric
+ * equation of a line a + t * v, where t = 1. The line enters and exits the
+ * interval at the points a + t<sub>1</sub> * v and a + t<sub>2</sub> * v
+ * respectively.
+ *
+ * @author Richard Domander
+ */
+@Plugin(type = Op.class)
+public class BoxIntersect extends
+	AbstractBinaryFunctionOp<ValuePair<Tuple3d, Vector3d>, Interval, Optional<ValuePair<DoubleType, DoubleType>>>
+	implements Contingent
+{
+
+	/**
+	 * If true, the method solves for intersections with the interval at
+	 * [minBounds, maxBounds). If false, then for maxBounds exactly.
+	 * <p>
+	 * For example, if you wanted to find the intersection points of a
+	 * [100x100x100] image stack, exclusion makes sure that ((50, 50, 0); (0, 0,
+	 * 1)) intersects the stack at (50, 50, 0) and (50, 50, 99), when the
+	 * coordinates are floored to integers. Thus, you won't get index out of
+	 * bounds exceptions.
+	 * </p>
+	 */
+	@Parameter(required = false)
+	private boolean excludeMaxBounds = true;
+
+	/**
+	 * Finds intersection points of the line and the minimum and maximum bounds of
+	 * the interval.
+	 *
+	 * @param line parametric equation of a line a + v as a (point, vector) pair.
+	 * @param interval an interval with integer coordinates.
+	 * @return scalar values (t<sub>1</sub>, t<sub>2</sub>) for intersection
+	 *         points (a + t<sub>1</sub> * v, a + t<sub>2</sub> * v).
+	 *         {@link Optional#empty()} if the line doesn't intersect the stack.
+	 */
+	@Override
+	public Optional<ValuePair<DoubleType, DoubleType>> calculate(
+		final ValuePair<Tuple3d, Vector3d> line, final Interval interval)
+	{
+		final int d = interval.numDimensions();
+		final long[] minBounds = new long[d];
+		interval.min(minBounds);
+		final long[] maxBounds = new long[d];
+		interval.max(maxBounds);
+		final Vector3d direction = new Vector3d(line.b);
+		final double length = direction.length();
+		if (length > 0.0 && Double.isFinite(length)) {
```

Comment:
throw IllegalArgumentException instead of this check?
I don't think there is a meaningful way of dealing with infinite entries in the vectors.

Reply:
Fixed in f379f79a5dc3ba904869c53e259aad86ac1b0107